### PR TITLE
eliminate some of the test suite warnings

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: pvlib/clearsky.py [hunk 1]
```diff
@@ -771,7 +777,7 @@ def rmse(alpha):
                       % max_iterations, RuntimeWarning)
 
     # be polite about returning the same type as was input
-    if isinstance(measured, pd.Series):
+    if ispandas:
```
Comment (CameronTStark) [inline pvlib/clearsky.py]: What's the advantage in declaring `ispandas` and only using it once? To be more idiomatic than `isinstance()`?

Comment (wholmgren) [reply pvlib/clearsky.py]: the line introduced above, `measured = np.asarray(measured)`, would make this `isinstance` always evaluate `False`. So I moved the check above the reassignment. I also thought of doing something like `measured_ = np.asarray(measured)`. Modifying the check resulted in a smaller diff, though.

**Comment by wholmgren** [comment]
There are still some windows warnings emitted in `losses` and `tracking`. I don't want to debug them via the CI, so I'd prefer to wrap this up and let someone else take those on.

File: pvlib/clearsky.py (hunk 1 above)
Comment (CameronTStark) [reply pvlib/clearsky.py]: I get it now. Thanks for clarifying that.

**Comment by CameronTStark** [comment]
> There are still some windows warnings emitted in `losses` and `tracking`. I don't want to debug them via the CI, so I'd prefer to wrap this up and let someone else take those on.

No problem. I'm glad to have a few less distracting warnings. I'll merge now. Thanks!

**Pull request merged**